src/Uno.UI/Extensions/PermissionsHelper.Android.cs

```diff
@@ -19,11 +19,185 @@ public static void Initialize()
+			return allPermissions;
+		}
+
+		private static List<string> PermissionToAsk(IList<string> allPermissions)
+		{// return list of permission, with all permission from input list that are not already granted
+			Android.Content.Context context = Android.App.Application.Context;
+			var askForPermission = new List<string>();
+
+			// check if permission is granted
+			foreach (var permission in allPermissions)
+			{
+				if (Android.Support.V4.Content.ContextCompat.CheckSelfPermission(context, permission)
+						!= Android.Content.PM.Permission.Granted)
+				{
+					askForPermission.Add(permission);
+				}
+			}
+			return askForPermission;
+		}
+
+
+		#endregion
+
+		/// <summary>
+		/// Return null on error, or array of permission to be asked for (not granted at this time). Both parameters can be null
+		/// </summary>
+		/// <param name="requiredPermissions">Permissions that are required. Can be null.</param>
+		/// <param name="optionalPermissions">Permissions that are required only if declared in Manifest. Can be null.</param>
+		/// <returns>Array of all permission that is not granted at this moment (can be empty!), or null if some error occured</returns>
+		public static string[] MissingPermissions(string[] requiredPermissions, string[] optionalPermissions)
+		{
+			// since API 29, we should do something more:
+			// https://developer.android.com/reference/android/content/pm/PackageInstaller.SessionParams.html#setWhitelistedRestrictedPermissions(java.util.Set%3Cjava.lang.String%3E)
+
+			// do we have declared permissions in Manifest?
+			var manifestPermissions = GetManifestPermissions();
+			if (manifestPermissions is null)
+			{
+				return null;
+			}
+
+
+			// test required permissions
+			if (!AreAllPermissionDeclared(manifestPermissions, requiredPermissions))
+			{
+				return null;
+			}
+
+			// prepare list of all permissions (all required, and all optional defined in Manifest
+			var allPermissions = PermissionWeWant(manifestPermissions, requiredPermissions, optionalPermissions);
+
+			// prepare list of permission to ask for
+			var askForPermission = PermissionToAsk(allPermissions);
+
+			return askForPermission.ToArray();
+
+		}
+
+		/// <summary>
+		/// Return true if granted are: all requiredPermissions, and all optionalPermissions mentioned in Manifest. Show dialog asking for missing permissions. Both parameters can be null.
+		/// </summary>
+		/// <param name="requiredPermissions">Permissions that are required. Can be null.</param>
+		/// <param name="optionalPermissions">Permissions that are required only if declared in Manifest. Can be null.</param>
+		/// <param name="ignoreErrors">if 'true', then do not throw exception on error, and simply retur false</param>
+		/// <returns>Bool if all permissions are granted, false if any of them is not granted</returns>
+
+		private static async Task<bool> TryAskPermissionAsync(CancellationToken ct, string[] requiredPermissions, string[] optionalPermissions, bool ignoreErrors)
```

Comment:
Let's use the same naming android uses.

```suggestion
		private static async Task<bool> TryRequestPermissionsAsync(CancellationToken ct, string[] requiredPermissions, string[] optionalPermissions, bool ignoreErrors)
```